src/Range.ts

```diff
@@ -0,0 +1,53 @@
+import SmartRange from "./SmartRange";
+
+const handler: ProxyHandler<SmartRange> = {
+    has(target, prop) {
+        if (typeof prop !== "symbol") {
+            const key = +prop;
+            if (!isNaN(key)) return target.includes(key);
+        }
+        return Reflect.has(target, prop);
+    },
+    set(target, prop, value) {
+        return Reflect.set(target, prop, value);
+    },
+    get(target, prop) {
+        if (prop === Symbol.iterator)
+            return target[Symbol.iterator].bind(target);
+        if (typeof prop !== "symbol") {
+            const key = +prop;
+            if (!isNaN(key)) return target.at(key);
+        }
+        return Reflect.get(target, prop);
+    },
+    // TODO ownKeys(target) {},
```

Comment:
TODOs. Should these stay in code?

Reply:
Yeah issues would probably be cleaner, but I expect that I'll resolve those TODO's before any other person gets a look at this project :D